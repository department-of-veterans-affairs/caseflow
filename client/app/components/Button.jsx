import React, { PropTypes } from 'react';
import { loadingSymbolHtml } from './RenderFunctions.jsx';

export default class Button extends React.Component {
  render() {
    let {
      classNames,
      id,
      name,
      disabled,
      loading,
      onClick,
      type
    } = this.props;

    if (!classNames) {
      classNames = ['cf-submit'];
    }

    if (disabled) {
      classNames.push('usa-button-disabled');
    }

    return <span>
    {loading && loadingSymbolHtml()}
    {!loading &&
      <button
        id={id || `${type}-${name.replace(/\s/g, '-')}`}
        className={classNames.join(' ')}
        type={type}
        disabled={disabled}
        onClick={onClick}>
          {name}
      </button>
    }
    </span>;
  }
}

Button.defaultProps = {
  type: 'button'
};

Button.propTypes = {
  classNames: PropTypes.arrayOf(PropTypes.string),
  disabled: PropTypes.bool,
  id: PropTypes.string,
  linkStyle: PropTypes.bool,
  loading: PropTypes.bool,
  name: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  type: PropTypes.string
};
