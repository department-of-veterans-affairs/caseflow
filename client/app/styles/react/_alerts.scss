
.usa-alert {

  &.fixed {
    position: fixed;
    top: 0;
    margin-top: 0;
    z-index: 999;
    left: 50%;
    transform: translate(-50%, 0);
  }

  // Remove after USWDS V.2.0
  &.usa-alert-slim {
    background-size: 4rem;

    .usa-alert-body {
      padding-left: 3.5rem;
    }

    .usa-alert-text {
      font-size: 2rem;
    }
  }

  .usa-alert-text {
    line-height: 1.5;
  }
}
