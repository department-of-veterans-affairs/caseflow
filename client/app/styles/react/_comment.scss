.comment-control-button-container {
  display: flex;
  width: 100%;
  padding: 3px 0;
  justify-content: space-between;
}

.comment-save-button-container {
  display: inline-block;
  width: 100%;
  margin-top: 5px;
  margin-bottom: 17px;
}

.comment-control-button {
  padding: 0;

  &.horizontal {
    text-align: left;
  }
}

.comment-control-button-divider {
  padding: 0 10px;
}

.comment-container-selected {
  box-shadow: 3px 3px 5px $color-gray-light;
}

.comment-container {
  word-wrap: break-word;
  word-break: break-word;
  background-color: $color-white;
  padding: 15px;
  cursor: pointer;

  &:not(.comment-horizontal-container) {
    margin-bottom: 15px;
    border-radius: 3px;
    border: 1px solid $color-gray-lighter;
  }

  &.comment-horizontal-container {
    padding: 25px 25px 25px 40px;
  }
}

// Max width is enforced so the size of this comment box
// fits with existing textareas in style guide
.comment-size-container {
  max-width: 430px;

  .cf-form-textarea {
    margin-bottom: 20px;
  }
}

.comment-horizontal-container {
  display: inline-block;
  width: 100%;
}

.comment-horizontal-spacing {
  margin-right: 49px;
}

.commentIcon-container {
  position: absolute;
  z-index: 10;
  // These dimensions need to match the comment icon SVG dimensions.
  width: 40px;
  height: 40px;
}

