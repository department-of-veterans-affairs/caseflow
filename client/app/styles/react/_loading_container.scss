.loadingContainer-positioning {
  position: absolute;
  width: 100%;
  height: 100%;
}

.loadingContainer-table {
  width: 100%;
  height: 100%;
  display: table;
}

.loadingContainer-table-cell {
  text-align: center;
  display: table-cell;
  vertical-align: middle;
}

.loadingContainer-content {
  position: relative;
  z-index: 1;

  iframe {
    width: 100%;
    height: 80vh;

    &.form9Viewer {
      height: 500px;
    }
  }
}

.loadingContainer-container {
  position: relative;
  overflow: hidden;
}

.cf-loading-button-text {
  display: inline-block;
  margin-right: 10px;
  line-height: 0;
  position: relative;
}

.cf-loading-message {
  .loading-message-text {
    display: inline-block;
    font-style: italic;
    font-weight: bolder;
  }

}

.cf-loading-button-symbol {
  display: inline-block;
}
