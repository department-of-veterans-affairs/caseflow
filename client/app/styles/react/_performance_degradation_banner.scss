.usa-banner {
  background-color: $color-gold-lightest;
  text-align: center;
}

.usa-banner-inner {
  padding: .8em;
}

.banner-icon {
  display: inline-block;
  vertical-align: text-top;
  // height: 25px;
}

.banner-text {
  padding-left: 6px;
}
