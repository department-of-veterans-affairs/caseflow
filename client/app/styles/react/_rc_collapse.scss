@mixin common() {

  width: 0;
  height: 0;
  font-size: 0;
  line-height: 0;
}

@mixin right($w, $h, $color) {

  border-top: $w solid transparent;
  border-bottom: $w solid transparent;
  border-left: $h solid $color;
}

@mixin bottom($w, $h, $color) {

  border-left: $w solid transparent;
  border-right: $w solid transparent;
  border-top: $h solid $color;
}

// scss-lint:disable IdSelector
#arrow {
  @include common();
  @include right(3px, 4px, $color-gray);
  @include bottom(3px, 4px, $color-gray);
}

.rc-collapse {

  &-anim-active {
    transition: height .2s ease-out;
  }

  & > &-item {
    &:first-child {
      border-top: none;
    }
  }

  & > &-item-disabled > .rc-collapse-header {
    cursor: not-allowed;
    color: $color-gray-light;
    background-color: $color-gray-lighter;
  }

  &-content {
    overflow: hidden;

    &-inactive {
      display: none;
    }
  }

  &-item:last-child {
    > .rc-collapse-content {
      border-radius: 0 0 3px 3px;
    }
  }

  & > &-item-active {
    > .rc-collapse-header {

      .arrow {

        #arrow > {
          @include bottom(3px, 4px, $color-gray);
        }
      }
    }
  }
}

.usa-accordion-content {
  &:not([aria-hidden]) {
    position: relative;
    left: 0;
  }
}
