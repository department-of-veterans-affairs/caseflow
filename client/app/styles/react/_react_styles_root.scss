@import '_comment.scss';
@import '_search_bar.scss';
@import '_performance_degradation_banner.scss';
@import '_rc_collapse.scss';
@import '_loading_container.scss';
@import '_select-dropdown.scss';
@import '_toggle_button.scss';
