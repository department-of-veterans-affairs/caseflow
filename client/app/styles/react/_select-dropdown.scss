$cf-select-bg-color: #fff;
$cf-select-disabled-bg-color: $color-gray-lightest;
$cf-select-disabled-color: #323a45;
$cf-select-border-color: #5b616b;
$cf-select-text-color: #323a45;

.cf-select {
  .cf-select__control {
    background-color: $cf-select-bg-color;
    border: 1px solid $cf-select-border-color;
    border-collapse: separate;
    border-radius: 0;
    border-spacing: 0;
    color: $cf-select-text-color;
    cursor: default;

    &--is-disabled {
      background-color: $cf-select-disabled-bg-color;

      .cf-select__single-value {
        color: $cf-select-disabled-color;
      }

      .cf-select__indicators {
        opacity: 0.35;
      }

      input {
        margin: 0;
      }
    }

    .cf-select__value-container {
      padding-bottom: 0;
      padding-top: 0;

      .cf-select__placeholder {
        color: $cf-select-text-color;
      }

      // Ensure that the "selected" value is aligned correctly
      .cf-select__placeholder,
      .cf-select__single-value {
        position: unset;
        top: unset;
        transform: unset;
      }
    }

    .cf-select__indicators {
      .cf-select__indicator-separator {
        background-color: $cf-select-border-color;
        margin-bottom: 0;
        margin-top: 0;
      }

      .cf-select__indicator {
        svg {
          fill: $cf-select-text-color;
        }
      }
    }
  }

  .cf-select__menu {
    border-radius: 0;

    .cf-select__menu-list {
      .cf-select__option {
        background-color: $color-white;
        color: $color-gray;

        &.cf-select__option--is-focused {
          background-color: $color-gray;
          color: $color-white;
        }
      }
    }
  }

  .usa-input-error & {
    .cf-select--is-disabled {
      input { width: 1px; }
    }
  }
}
