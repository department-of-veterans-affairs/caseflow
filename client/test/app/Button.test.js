import React from 'react';
import { render, screen } from '@testing-library/react';
import { axe } from 'jest-axe';
import userEvent from '@testing-library/user-event';

import Button from '../../app/components/Button';

describe('Button', () => {
  const defaults = {
    name: 'test-button',
    ariaLabel: 'test-button-aria',
    children: 'Click me'
  };

  const setup = (props) => {
    const { container } = render(<Button {...defaults} {...props} />);

    return { container };
  };

  it('renders properly', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  it('passes a11y testing', async () => {
    const { container } = setup();

    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });

  it('calls onClick when clicked', async () => {
    const onClick = jest.fn();

    setup({ onClick });

    const buttonElement = screen.getByRole('button');

    expect(buttonElement).toBeInTheDocument();
    expect(onClick).toHaveBeenCalledTimes(0);

    await userEvent.click(buttonElement);
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('respects loading prop', () => {
    const loading = true;

    setup({ loading });

    const loadingText = 'Loading...';
    const loadingTextElement = screen.getByText(loadingText);

    expect(loadingTextElement).toBeInTheDocument();
  });
  it('respects disabled prop and removes other classes when disabled', () => {
    const disabled = true;

    setup({ disabled });

    const buttonElement = screen.getByRole('button');

    expect(buttonElement.classList.contains('usa-button-disabled')).toBeTruthy();
    expect(buttonElement.classList.contains('usa-button-primary')).toBeFalsy();
  });
});
