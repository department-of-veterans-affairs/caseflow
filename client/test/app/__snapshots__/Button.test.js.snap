// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders properly 1`] = `
<div>
  <span>
    <button
      aria-label="test-button-aria"
      class="cf-submit usa-button"
      id="button-test-button"
      type="button"
    >
      Click me
    </button>
  </span>
</div>
`;
